overlord/devicestate/firstboot.go

```diff
@@ -129,93 +129,107 @@ func populateStateFromSeedImpl(st *state.State, tm timings.Measurer) ([]*state.T
 
 	tsAll := []*state.TaskSet{}
 	configTss := []*state.TaskSet{}
+	chainTs := func(all []*state.TaskSet, ts *state.TaskSet) []*state.TaskSet {
+		n := len(all)
+		if n != 0 {
+			ts.WaitAll(all[n-1])
+		}
+		return append(all, ts)
+	}
 
 	baseSnap := "core"
+	classicWithSnapd := false
 	if model.Base() != "" {
 		baseSnap = model.Base()
 	}
+	if _, ok := seeding["snapd"]; release.OnClassic && ok {
```

Comment:
shouldn't this go before the check for `model.Base()` in case you're on classic with a model with a base?

Reply:
classic models can't have a base